### PR TITLE
DEP Upgrade frontend build stack

### PR DESCRIPTION
**Pull request by GuySartorelli**
See https://github.com/silverstripe/silverstripe-admin/pull/1389 for a wider discussion on most of the changes

## Parent issue
- https://github.com/silverstripeltd/product-issues/issues/644

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,24 +17,21 @@
...
   "license": "BSD-3-Clause",
+  "engines": {
+    "node": "^18.x"
+  },
   "dependencies": {
-    "classnames": "^2.2.5",
-    "jquery": "^3.5.0",
+    "classnames": "^2.3.2",
     "merge": "^2.1.1",
-    "mime": "^1.4.1",
```
Comment (ChloeSartorelli) [inline package.json:24]: mime isn't used

File: package.json [hunk 2]
```diff
@@ -17,24 +17,21 @@
...
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "reactstrap": "^8.9.0"
   },
   "devDependencies": {
-    "@silverstripe/eslint-config": "^0.0.5",
-    "@silverstripe/webpack-config": "^1.2.0",
-    "babel-jest": "^20.0.3",
-    "bootstrap": "^4.3.1",
-    "jest-cli": "^19.0.2"
```
Comment (ChloeSartorelli) [inline package.json:35]: Neither jest nor bootstrap is used

File: client/src/components/ColorPickerField/ColorPickerField.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ class ColorPickerField extends Component {
   }
 }
 
-ColorPickerField.proptypes = {
+ColorPickerField.propTypes = {
```
Comment (ChloeSartorelli) [inline client/src/components/ColorPickerField/ColorPickerField.js:114]: Fixes `error  Typo in static class property declaration  react/no-typos` (see [the docs on this rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-typos.md))

**Comment by GuySartorelli** [comment]
This module includes a tinymce4 fontawesome4 plugin in a thirdparty directory. This is wildly outdated, and [the source code on github](https://github.com/josh18/TinyMCE-FontAwesome-Plugin) isn't much better.
I've found a few tinymce5 plugins that provide similar functionality but nothing for tinymce 6.

We'll need to either:
1. Fork an existing tinymce fontawesome plugin and update it to be compatible with tinymce 6
     - not the one in thirdparty 'cause that's way too old now - but there are some that are compatible with tinymce 5 which might be easy-ish to upgrade
1. Make our own tinymce fontawesome plugin and maintain it
1. Ditch the fontawesome tinymce plugin for this module

We should also add behat tests for all of the js functionality in this module.

File: _config.php [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 use SilverStripe\Core\Environment;
-use SilverStripe\Forms\HTMLEditor\HTMLEditorConfig;
 use SilverStripe\Core\Manifest\ModuleLoader;
+use SilverStripe\Forms\HTMLEditor\TinyMCEConfig;
 
 // By default the FontAwesome plugin for TinyMCE is enabled. You can disable it by defining
 // CWP_AGENCY_DISABLE_FONTAWESOME_PLUGIN = true in your environment configuration.
 if (!Environment::getEnv('CWP_AGENCY_DISABLE_FONTAWESOME_PLUGIN')) {
-    $cwpEditor = HtmlEditorConfig::get('cwp');
+    $cwpEditor = TinyMCEConfig::get('cwp');
```
Comment (ChloeSartorelli) [inline _config.php]: `HtmlEditorConfig` doesn't actually have some of the methods that are being called - this is _explicitly_ for tinymce configuration.

File: client/src/components/ColorPickerField/ColorPickerField.scss [hunk 5]
```diff
@@ -46,4 +43,9 @@
     align-self: center;
     margin-left: 12px;
   }
+
+  .popover {
+    max-width: 380px;
+    width: 380px;
+  }
```
Comment (ChloeSartorelli) [inline client/src/components/ColorPickerField/ColorPickerField.scss:50]: It looks like whatever controls the underlying popover has changed its markup - this is now a div below the `.color-picker-field-popover` div. If we keep it as it was, there is 380px of whitespace to the right of the form any time you open one of the dropdowns.

**Comment by maxime-rainville** [comment]
> This module includes a tinymce4 fontawesome4 plugin in a thirdparty directory. This is wildly outdated, and [the source code on github](https://github.com/josh18/TinyMCE-FontAwesome-Plugin) isn't much better.

My assumption here is that whoever includes FA icons in a WYSIWYG probably use them effectively as emojis. When FA first came on the scene, Emojis were not as widely supported as they are today. That's probably why no one bothered porting those those FA plugins to work with TinyMCE6.

TinyMCE6 has a [free Emoticon plugin](https://www.tiny.cloud/docs/plugins/opensource/emoticons/).

My suggested approach at this stage is:
- Remove the FA plugin on CMS6.
- Maybe add some extra logic just to make sure that existing FA icon still show in the HTMLEditor and in the front end.
- Maybe enable the Emoticon plugin in the CWP Agency Extension module.

**Comment by GuySartorelli** [comment]
I will remove the legacy font-awesome plugin.

I will not enable the emoticon plugin, as I see that as being for a fundamentally different use-case (there aren't even any emoticons/emojis in font-awesome - see the icons in [the icons list](https://fontawesome.com/v4/icons/) compared with [the tinymce emoji](https://www.tiny.cloud/docs/plugins/opensource/emoticons/) - there's not a lot of crossover other than I guess the objects category)
I will call it out as a potential alternative that developers can enable if they think it is suitable for their project.

There is an existing `CWP_AGENCY_DISABLE_FONTAWESOME_PLUGIN` environment variable which, if set, disables the plugin. I'm going to flip this on its head and replace it with `CWP_AGENCY_ENABLE_FONTAWESOME_STYLES`.
If the new variable is set, then the `extended_valid_elements` configuration and font awesome css (from a cdn) will be added to the tinymce config like it used to be. If that is not set, then no font awesome configuration or css will be added at all.

You already had to add css on the front-end separately, so that will not be affected. Projects using this module which wish to keep any existing font-awesome icons in their content will need to set `CWP_AGENCY_ENABLE_FONTAWESOME_STYLES` to true in their environment variables to see the icons in the CMS WYSIWYG, and to ensure the icons are retained when saving.

Note that is would not be feasible to set this to a yaml config since it requires calling methods on the `TinyMCEConfig` in `_config.php`, which happens before some yaml config is set. If there are any projects which _cannot_ set environment variables, they can set the appropriate config and inject the css themselves for their project, which is trivial.

I will call all of this out in the changelog.

**Review by emteknetnz — APPROVED** [review]
Tested locally

**Pull request merged**